src/CommunityToolkit.Maui/Behaviors/PlatformBehaviors/Touch/GestureManager.cs

```diff
@@ -0,0 +1,789 @@
+using System.Text;
+using System.Threading.Tasks;
+using CommunityToolkit.Maui.Extensions;
+using static System.Math;
+
+namespace CommunityToolkit.Maui.Behaviors;
+
+sealed class GestureManager : IDisposable
+{
+	const int animationProgressDelay = 10;
+
+	Color? defaultBackgroundColor;
+
+	CancellationTokenSource? longPressTokenSource;
+
+	CancellationTokenSource? animationTokenSource;
+
+	Func<TouchBehavior, TouchState, HoverState, int, Easing?, CancellationToken, Task>? animationTaskFactory;
+
+	double? durationMultiplier;
+
+	double animationProgress;
+
+	TouchState animationState;
+
+	internal void HandleTouch(TouchBehavior sender, TouchStatus status)
+	{
+		if (sender.IsDisabled)
+		{
+			return;
+		}
+
+		var canExecuteAction = sender.CanExecute;
+		if (status != TouchStatus.Started || canExecuteAction)
+		{
+			if (!canExecuteAction)
+			{
+				status = TouchStatus.Canceled;
+			}
+
+			var state = status == TouchStatus.Started
+				? TouchState.Pressed
+				: TouchState.Normal;
+
+			if (status == TouchStatus.Started)
+			{
+				animationProgress = 0;
+				animationState = state;
+			}
+
+			var isToggled = sender.IsToggled;
+			if (isToggled.HasValue)
+			{
+				if (status != TouchStatus.Started)
+				{
+					durationMultiplier = (animationState == TouchState.Pressed && !isToggled.Value) ||
+						(animationState == TouchState.Normal && isToggled.Value)
+						? 1 - animationProgress
+						: animationProgress;
+
+					UpdateStatusAndState(sender, status, state);
+
+					if (status == TouchStatus.Canceled)
+					{
+						sender.ForceUpdateState(false);
+						return;
+					}
+
+					OnTapped(sender);
+					sender.IsToggled = !isToggled;
+					return;
+				}
+
+				state = isToggled.Value
+					? TouchState.Normal
+					: TouchState.Pressed;
+			}
+
+			UpdateStatusAndState(sender, status, state);
+		}
+
+		if (status == TouchStatus.Completed)
+		{
+			OnTapped(sender);
+		}
+	}
+
+	internal void HandleUserInteraction(TouchBehavior sender, TouchInteractionStatus interactionStatus)
+	{
+		if (sender.InteractionStatus != interactionStatus)
+		{
+			sender.InteractionStatus = interactionStatus;
+			sender.RaiseInteractionStatusChanged();
+		}
+	}
+
+	internal void HandleHover(TouchBehavior sender, HoverStatus status)
+	{
+		if (!sender.Element?.IsEnabled ?? true)
+		{
+			return;
+		}
+
+		var hoverState = status == HoverStatus.Entered
+			? HoverState.Hovered
+			: HoverState.Normal;
+
+		if (sender.HoverState != hoverState)
+		{
+			sender.HoverState = hoverState;
+			sender.RaiseHoverStateChanged();
+		}
+
+		if (sender.HoverStatus != status)
+		{
+			sender.HoverStatus = status;
+			sender.RaiseHoverStatusChanged();
+		}
+	}
+
+	internal async Task ChangeStateAsync(TouchBehavior sender, bool animated)
+	{
+		var status = sender.Status;
+		var state = sender.State;
+		var hoverState = sender.HoverState;
+
+		AbortAnimations(sender);
+		animationTokenSource = new CancellationTokenSource();
+		var token = animationTokenSource.Token;
```

Comment:
you create this local variable but you use `animationTokenSource.Token` when passing the token to other methods, like at line `153`. Can you refactor to pass the local variable instead?

Reply:
This has been updated to not use the local variable